jujugui/static/gui/src/app/app.js

```diff
@@ -1320,6 +1238,9 @@ YUI.add('juju-gui', function(Y) {
           serviceName: 'charmstore'
         });
         this.set('charmstore', new Charmstore(charmstoreURL, apiPath, bakery));
+        // Store away the charmstore auth info.
+        var macaroon = bakery.getMacaroon();
```

Comment:
This will only work if the macaroon is already stored in the users browser on load because the fetching of the macaroon is async. Maybe this method should be passed into the bakery.